File: desafio1-1/desafio1-1/ViewController.m
//
//  ViewController.m
//  Desafio01
//
//  Created by Pedro Almeida on 05/08/21.
//

#import "ViewController.h"
#import "Desafio2ViewController.h"

@interface ViewController ()

@end

@implementation ViewController




- (void)viewDidLoad {
    [super viewDidLoad];
    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(abrirControler) ];
    [_btn1 addGestureRecognizer:tap];
    NSLog(@"CARREGUEI");
}
-(void) abrirControler{
    Desafio2ViewController *d = [[Desafio2ViewController alloc] initWithNibName:@"Desafio2ViewController" bundle:nil];

    [self.navigationController pushViewController: d animated:NO];
   
}


- (IBAction)changeView:(UIButton *)sender {
    NSLog(@"funciona");
    [self abrirControler];
}


- (IBAction)showAlert:(id)sender {
    [self ok1];
}

-(void)ok1:(NSString*)	{
    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Errado"
                                   message:@"Tenta mais uma vez"
                      preferredStyle:UIAlertControllerStyleAlert];

    UIAlertAction* ok = [UIAlertAction actionWithTitle:@"Repetir alerta" style:UIAlertActionStyleDefault
        handler:^(UIAlertAction * action) {
        NSLog(@"ok");
        [self ok1];

    }];
    UIAlertAction* cancelar = [UIAlertAction actionWithTitle:@"Sair do Alert" style:UIAlertActionStyleCancel handler:nil];

    [alert addAction:ok];
    [alert addAction:cancelar];

    [self presentViewController:alert animated:YES completion:nil];
}
@end

File: desafio1.swift/desafio1.swift/ViewController.swift
//
//  ViewController.swift
//  desafio1.swift
//
//  Created by Pedro Almeida on 09/08/21.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    @IBAction func ShowAlert(_ sender: Any) {
        self.createAlert()
    }
    func createAlert(){
        let alert = UIAlertController(title: "errado", message: "Você escolheu o caminho errado amigo", preferredStyle: UIAlertController.Style.alert)
        alert.addAction(UIAlertAction(title: "Repetir alert", style: UIAlertAction.Style.default, handler: { ACTION in
            print("ok")
            self.createAlert()
        }))
        alert.addAction(UIAlertAction(title: "Cancelar", style: UIAlertAction.Style.cancel, handler: nil))
        self.present(alert, animated: true, completion: nil)

    }
    
    @IBAction func changeView(_ sender: Any) {
        self.instantiateView()
    }
    func instantiateView() {
        let desafio1ViewController =  desafio1ViewController(nibName: nil, bundle: nil)
        self.navigationController?.pushViewController(desafio1ViewController, animated: true)
    }
}

